frontend/src/toolbar/button/ToolbarButton.js

```diff
@@ -0,0 +1,257 @@
+            const ns = y < window.innerHeight / 2 ? 's' : 'n'
+            const we = x < window.innerWidth / 2 ? 'e' : 'w'
+            // use toolbarButtonLogic.values.quarter to always get the last state
+            if (toolbarButtonLogic.values.quarter !== `${ns}${we}`) {
+                setQuarter(`${ns}${we}`)
+            }
+        }
+    }
+
+    useEffect(() => updateQuarter(), [])
+
+    const longPressEvents = useLongPress(
+        clicked => {
+            updateQuarter()
+            if (isAuthenticated) {
+                if (clicked) {
+                    dock()
+                } else {
+                    // extendButtons()
+                }
+            } else {
+                authenticate()
+            }
+        },
+        { ms: 700, clickMs: 1 }
+    )
+
+    const globalMouseMove = useRef(null)
+    useEffect(() => {
+        globalMouseMove.current = function(e) {
+            const buttonDiv = getShadowRoot()?.getElementById('button-toolbar')
+            if (buttonDiv) {
+                updateQuarter(buttonDiv)
+                const rect = buttonDiv.getBoundingClientRect()
+                const x = rect.left + rect.width / 2
+                const y = rect.top + rect.height / 2
+                const distance = Math.sqrt((e.clientX - x) * (e.clientX - x) + (e.clientY - y) * (e.clientY - y))
+
+                const startDistance = isAuthenticated ? 230 : 130
+                const endDistance = isAuthenticated ? 160 : 60
+
+                if (distance >= startDistance) {
+                    if (toolbarButtonLogic.values.extensionPercentage !== 0) {
+                        setExtensionPercentage(0)
+                    }
+                } else if (distance >= endDistance && distance < startDistance) {
+                    setExtensionPercentage((startDistance - distance) / (startDistance - endDistance))
+                } else if (distance < endDistance) {
+                    if (toolbarButtonLogic.values.extensionPercentage !== 1) {
+                        setExtensionPercentage(1)
+                    }
+                }
+            }
+        }
+        window.addEventListener('mousemove', globalMouseMove.current)
+        return () => window.removeEventListener('mousemove', globalMouseMove.current)
+    }, [isAuthenticated])
+
+    let index = 0
+    const itemCount = 3
+    const padding = -20
+    const distance = extensionPercentage * 100
+    const closeDistance = extensionPercentage * 50
+    const inspectDistance = inspectingElement || selectedInspectElement ? Math.max(50, distance) : distance
+    const heatmapDistance = heatmapEnabled ? Math.max(50, distance) : distance
+
+    return (
+        <Circle
+            rootNode
+            width={64}
+            className="floating-toolbar-button"
+            content={<Logo style={{ width: 54, height: 54, filter: 'invert(1)', cursor: 'pointer' }} />}
+            label={isAuthenticated ? 'Toolbar' : 'Oh, Hello!'}
```

Comment:
"Oh, Hello" could be a bit more information. (Click to use the toolbar) or something

Reply:
I added an interactive text element there. With a monospace font and the new logo it will probably feel quite 80s if needed...